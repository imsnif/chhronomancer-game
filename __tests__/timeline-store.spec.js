import moment from 'moment'
import renderer from 'react-test-renderer'

beforeEach(() => {
  jest.resetModules()
})

test('TimelineStore => is constructed properly', () => {
  const timelineStore = require('../stores/timeline').default
  const timelines = timelineStore.timelines.map(p => p)
  const { sortBy } = timelineStore
  expect(timelines).toEqual([])
})

test('TimelineStore => can add a timeline', () => {
  const timelineStore = require('../stores/timeline').default
  timelineStore.addTimeline('Timeline 1', 'steal')
  const timelines = timelineStore.timelines.map(p => p)
  expect(timelines).toMatchSnapshot()
})

test('TimelineStore => can add a player to a timeline', () => {
  const timelineStore = require('../stores/timeline').default
  timelineStore.addTimeline('Timeline 1', 'steal')
  const timelines = timelineStore.timelines.map(p => p)
  timelineStore.addPlayer('Timeline 1', 1)
  const { players } = timelineStore.getTimeline('Timeline 1')
  expect(players).toMatchSnapshot()
})

test('TimelineStore => can remove a player', () => {
  const timelineStore = require('../stores/timeline').default
  timelineStore.addTimeline('Timeline 1', 'steal')
  const timelines = timelineStore.timelines.map(p => p)
  timelineStore.addPlayer('Timeline 1', 1)
  timelineStore.removePlayer('Timeline 1', 1)
  const { players } = timelineStore.getTimeline('Timeline 1')
  expect(players).toMatchSnapshot()
})

test('TimelineStore => can change sort', () => {
  const timelineStore = require('../stores/timeline').default
  timelineStore.addTimeline('Timeline 1', 'steal')
  const timelines = timelineStore.timelines.map(p => p)
  timelineStore.changeSort('Timeline 1', 'type')
  const { sortBy } = timelineStore.getTimeline('Timeline 1')
  expect(sortBy).toEqual('type')
})
