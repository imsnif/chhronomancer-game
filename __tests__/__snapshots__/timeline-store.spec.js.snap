// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineStore => can add a player to a timeline 1`] = `
Array [
  1,
]
`;

exports[`TimelineStore => can add a timeline 1`] = `
Array [
  Object {
    "name": "Timeline 1",
    "players": Array [],
    "sortBy": "player",
    "type": "steal",
  },
]
`;

exports[`TimelineStore => can remove a player 1`] = `Array []`;
